apps/meteor/server/services/video-conference/service.ts

```diff
@@ -0,0 +1,186 @@
+	): Promise<string> {
+		const configs = [
+			`userInfo.displayName="${user.name}"`,
+			`config.prejoinPageEnabled=false`,
+			`config.requireDisplayName=false`,
+			`config.callDisplayName="Direct Message"`,
+		];
+
+		console.log(user.username, options);
+		if (options.mic !== undefined) {
+			configs.push(`config.startWithAudioMuted=${options.mic ? 'true' : 'false'}`);
+		}
+		if (options.cam !== undefined) {
+			configs.push(`config.startWithVideoMuted=${options.cam ? 'true' : 'false'}`);
+		}
+
+		const configHash = configs.join('&');
+		const url = `${call.url}#${configHash}`;
+
+		return url;
+	}
+}
+
+export class VideoConfService extends ServiceClassInternal implements IVideoConfService {
+	protected name = 'video-conference';
+
+	private Users: UsersRaw;
+
+	private Rooms: RoomsRaw;
+
+	private VideoConference: VideoConferenceRaw;
+
+	constructor(db: Db) {
+		super();
+
+		this.Users = new UsersRaw(db.collection('users'));
+		this.VideoConference = new VideoConferenceRaw(db.collection('rocketchat_video_conference'));
+		this.Rooms = new RoomsRaw(db.collection('rocketchat_room'));
+	}
+
+	public async start(caller: IUser['_id'], rid: string): Promise<VideoConferenceInstructions> {
+		const room = await this.Rooms.findOneById<Pick<IRoom, '_id' | 't' | 'uids'>>(rid, { projection: { t: 1, uids: 1 } });
+
+		if (!room) {
+			throw new Error('invalid-room');
+		}
+
+		if (room.t === 'd' && room.uids && room.uids.length <= 2) {
+			return this.startDirect(caller, room);
+		}
+
+		throw new Error('Conferece calls have not been implemented yet.');
+	}
+
+	public async join(uid: IUser['_id'], callId: IVideoConference['_id'], options: VideoConferenceJoinOptions): Promise<string> {
+		const call = await this.VideoConference.findOneById(callId);
+		if (!call) {
+			throw new Error('invalid-call');
+		}
+
+		const user = await this.Users.findOneById<Pick<IUser, '_id' | 'username' | 'name'>>(uid, { projection: { name: 1, username: 1 } });
+		if (!user) {
+			throw new Error('failed-to-load-own-data');
+		}
+
+		switch (call.type) {
+			case 'direct':
+				return this.joinDirect(call, user, options);
+		}
+
+		throw new Error('Conferece calls have not been implemented yet.');
+	}
+
+	public async get(callId: IVideoConference['_id']): Promise<IVideoConference | null> {
+		return this.VideoConference.findOneById(callId);
+	}
+
+	private async startDirect(caller: IUser['_id'], { _id: rid, uids }: AtLeast<IRoom, '_id' | 'uids'>): Promise<DirectCallInstructions> {
+		const callee = uids?.filter((uid) => uid !== caller).pop();
+		if (!callee) {
+			// Are you trying to call yourself?
+			throw new Error('invalid-call-target');
+		}
+
+		const user = await this.Users.findOneById<IUser>(caller, {});
+		if (!user) {
+			throw new Error('failed-to-load-own-data');
+		}
+
+		const callId = new ObjectID().toHexString();
+		// #ToDo: Generate the 'calling' message using the callId
+
+		await this.VideoConference.insertOne({
```

Comment:
an idea that is not standardized nor well defined, but what do you think of having a `create` method, so you provide the real data and it adds the "hardcoded" one?